### PR TITLE
Uar 1321 confirm removal statement

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1321

### Change description

Template
Routing
Navigation
Controller
Validation

Post function incomplete as pages to navigate to do not yet exist.
Also navigation to the page is not yet possible as this is dependent on previous page sin the journey being completed.
### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/update/remove.confirm.statement.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+jest.mock("../../../src/utils/logger");
```
Comment (markpit) [inline test/controllers/update/remove.confirm.statement.controller.spec.ts:1]: can you add jest.mock("ioredis");
at the top?
not adding it causes the 'test was forced to quit after timeout' message when you run the tests

File: test/controllers/update/remove.confirm.statement.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const mockGetRemove = getRemove as jest.Mock;
+
+describe("Remove confirmation statement controller", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockGetRemove.mockReturnValue({});
+  });
+
+  describe ("GET tests", () => {
+    test(`renders the ${config.REMOVE_CONFIRM_STATEMENT_PAGE}`, async () => {
```
Comment (markpit) [inline test/controllers/update/remove.confirm.statement.controller.spec.ts:38]: the appData is passed into the template and the entity name and number are rendered on the screen (taken from appData). Could you put dummy values into the mock appData and check those dummy values are in the resp.text

Comment (mattch1) [reply test/controllers/update/remove.confirm.statement.controller.spec.ts:44]: Created dummy values and checks fors them.

File: src/controllers/update/remove.confirm.statement.controller.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    });
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `POST ${config.REMOVE_CONFIRM_STATEMENT_PAGE}`);
+    return res.redirect(`${config.REMOVE_CANNOT_USE_URL}`);
```
Comment (mwestacott) [inline src/controllers/update/remove.confirm.statement.controller.ts:23]: Is this redirection just a temporary implementation? Until the next screen/story is done? If yes, might be worth making this clear with a TODO comment.

File: src/utils/registrable.beneficial.owner.ts [hunk 4]
```diff
@@ -47,13 +48,17 @@ export const postRegistrableBeneficialOwner = async (req: Request, res: Response
     setExtraData(req.session, appData);
     await saveAndContinue(req, session, false);
 
-    if (noChangeFlag) {
-      noChangeHandler(req, res);
+    if (isRemoveJourney(req)) {
+      return res.redirect(config.REMOVE_CONFIRM_STATEMENT_URL);
     } else {
```
Comment (mwestacott) [inline src/utils/registrable.beneficial.owner.ts:53]: I think that the `if (noChangeFlag)` check can go on this line, with no need for an extra conditional block.

File: test/__mocks__/text.mock.ts [hunk 5]
```diff
@@ -254,6 +254,7 @@ export const REMOVE_SOLD_ALL_LAND_FILTER_PAGE_TITLE = "Has the overseas entity d
 export const REMOVE_IS_ENTITY_REGISTERED_OWNER_TITLE = "Is the overseas entity currently listed on any land registry records as the registered owner of property or land in the UK?";
 export const REMOVE_INTERRUPT_CARD_TEXT = "Once an overseas entity is removed";
 export const REMOVE_USE_INFORMATION_NEED_MORE = "We’ll use this if we need more information about the application.";
+export const REMOVE_CONFIRMATION_STATEMENT_TITLE = "Confirm the removal statement - Apply to remove and overseas entity";
```
Comment (mwestacott) [inline test/__mocks__/text.mock.ts:257]: Text content here doesn't look right - 'and overseas entity'. Surely it should be 'an overseas entity'?

But if this is wrong and tests are passing, either it's not actually being used by a test OR it's also wrong somewhere else.

File: views/update/remove-confirm-statement.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
+{% extends "update-layout.html" %}
+
+{% set title = "Confirm the removal statement - Apply to remove and overseas entity" %}
```
Comment (mwestacott) [inline views/update/remove-confirm-statement.html:3]: Ah yes, so text looks wrong here too. Does it come from the prototype? Might need to clarify with analyst and UI team.

File: src/controllers/update/remove.confirm.statement.controller.ts (hunk 3 above)
Comment (mattch1) [reply src/controllers/update/remove.confirm.statement.controller.ts:24]: Added TODO

File: src/utils/registrable.beneficial.owner.ts (hunk 4 above)
Comment (mattch1) [reply src/utils/registrable.beneficial.owner.ts:61]: Rearranged this and added validation flag for not matching screen  - AC4

File: views/update/remove-confirm-statement.html (hunk 6 above)
Comment (mattch1) [reply views/update/remove-confirm-statement.html]: As above

**Comment by mattch1** [comment]
Added code for AC4 "the statement you have chosen does not match the information provided in this update"

File: src/utils/registrable.beneficial.owner.ts (hunk 4 above)
Comment (mwestacott) [reply src/utils/registrable.beneficial.owner.ts:61]: Nice.

**Review by mwestacott — APPROVED** [review]

File: src/utils/registrable.beneficial.owner.ts [hunk 7]
```diff
@@ -25,7 +26,7 @@ export const getRegistrableBeneficialOwner = (req: Request, res: Response, next:
     return res.render(templateName, {
       backLinkUrl: backLinkUrl,
       templateName: templateName,
-      appData,
+      ...appData,
```
Comment (markpit) [inline src/utils/registrable.beneficial.owner.ts:29]: the ...appData change here will also affect the update screen, I'm guessing the update version of this screen had been tested previously and working ok so this change might break it?

Comment (mattch1) [reply src/utils/registrable.beneficial.owner.ts:29]: We need this because the company name and number do not display when there are errors on the page.

**Review by markpit — APPROVED** [review]

**Pull request merged**